mithril-aggregator/src/dependency_injection/builder.rs

```diff
@@ -548,17 +548,14 @@ impl DependenciesBuilder {
 
     // TODO: cache management which should not be ASYNC
     async fn build_immutable_digester(&mut self) -> Result<Arc<dyn ImmutableDigester>> {
-        let immutable_digester_cache = match self.configuration.environment {
-            ExecutionEnvironment::Production => Some(self.get_immutable_cache_provider().await?),
-            _ => None,
-        };
-        let digester = CardanoImmutableDigester::new(
-            self.configuration.db_directory.clone(),
-            immutable_digester_cache,
-            self.get_logger().await?,
-        );
-
-        Ok(Arc::new(digester))
+        match self.configuration.environment {
+            ExecutionEnvironment::Production => Ok(Arc::new(CardanoImmutableDigester::new(
+                self.configuration.db_directory.clone(),
+                Some(self.get_immutable_cache_provider().await?),
+                self.get_logger().await?,
+            ))),
+            _ => Ok(Arc::new(DumbImmutableDigester::default())),
```

Comment:
This is not a neutral change, before you had a complete digester than compute real hash (but without cache) now you have a fake digester that does no computation, is this really what we want here ?